File: src/main/java/net/exoad/picflux/shared/RoutineScheduler.java
package net.exoad.picflux.shared;

import java.util.Timer;
import java.util.concurrent.Executors;
import java.util.concurrent.ScheduledExecutorService;
public final class RoutineScheduler
{
	private static RoutineScheduler instance;
	
	public static RoutineScheduler getInstance()
	{
		if(instance==null)
				instance=new RoutineScheduler();
		return instance;
	}
	
	public ScheduledExecutorService defaultPool;
	public Timer periodic;
	
	private RoutineScheduler()
	{
		defaultPool=
			Executors.newScheduledThreadPool(Runtime
														 .getRuntime()
														 .availableProcessors()*2);
		periodic=new Timer(true);
	}
}

File: src/main/java/net/exoad/picflux/shared/RoutineScheduler.kt
package net.exoad.picflux.shared

import java.util.Timer
import java.util.concurrent.Executors
import java.util.concurrent.ScheduledExecutorService

class RoutineScheduler
private constructor()
{
	var defaultPool:ScheduledExecutorService=Executors.newScheduledThreadPool(
		Runtime.getRuntime().availableProcessors()*2
	)
	var periodic:Timer=Timer(true)
	
	companion object
	{
		private var instance:RoutineScheduler?=null
		
		fun getInstance():RoutineScheduler
		{
			if (instance==null) instance=RoutineScheduler()
			return instance!!
		}
	}
}
